### DISCUSSION TITLE
Idea - building a web designing tool using react-flow

### DISCUSSION THREAD

**Discussion by ghost** [xyflow/xyflow#2846] [Ideas]
**Redacted
I'm changing user experience of my project https://github.com/imvetri/ui-editor from a GUI based to Node based. I'm trying to build a prototype design tool.

Ask: I wanted to write less code and build nodes, I thought if would be cool if I can create nodes, handles, backgrounds on the fly i.e create a tool box for react-flow's API in UI form. Before building it, I thought I'd drop a note here to ask if someone had done it.

**

Update - react-flow has all the features I was looking for to build a design tool
Examples
1. new node on edge drop
2. proximity connect
3. node resize
4. dynamic node

my tool has few features and if we can mix both and build a design to code tool. Feel free to let me know if it sounds interesting.

**Comment by ghost** [comment]
**Progress made** - Setup a project without npm, node, webpack. 
**Lesson learnt** - Thinking in tools versus, thinking to solve a problem.
**Similar ideas** - Getting rid of bloatware - elon musk.

**Reply by syntrydy** [reply]
Where is the code for contribution?

**Reply by levirtevs** [reply]
https://github.com/imvetri/ui-editor/blob/from.scratch/ui-editor.html

**Reply by ghost** [reply]
@syntrydy  here is the repo "https://github.com/imvetri/ui-editor" and this is the branch name "from.scratch" 

feel free to reach out to me if you need any help or questions or directions. I'l be glad to support.
